Calibration/PCCAlCaRecoProducers/plugins/AlcaPCCProducer.cc

```diff
@@ -0,0 +1,115 @@
+{
+    resetNLumi_ = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getUntrackedParameter<int>("resetEveryNLumi",-1);
+    fPixelClusterLabel = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getParameter<edm::InputTag>("pixelClusterLabel");
+    trigstring_ = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getUntrackedParameter<std::string>("trigstring","alcaPCC");
+
+    ftotalevents = 0;
+    countLumi_ = 0;
+    beginLumiOfPCC_ = endLumiOfPCC_ = -1;
+
+    produces<reco::PixelClusterCounts, edm::InLumi>(trigstring_);
+    pixelToken=consumes<edmNew::DetSetVector<SiPixelCluster> >(fPixelClusterLabel);
+}
+
+//--------------------------------------------------------------------------------------------------
+AlcaPCCProducer::~AlcaPCCProducer(){
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
+    ftotalevents++;
+
+    unsigned int bx=iEvent.bunchCrossing();
+    //std::cout<<"The Bunch Crossing"<<bx<<std::endl;
+    thePCCob->eventCounter(bx);
+
+    //Looping over the clusters and adding the counts up  
+    edm::Handle< edmNew::DetSetVector<SiPixelCluster> > hClusterColl;
+    iEvent.getByToken(pixelToken,hClusterColl);
+
+    const edmNew::DetSetVector<SiPixelCluster>& clustColl = *(hClusterColl.product()); 
+    // ----------------------------------------------------------------------
+    // -- Clusters without tracks
+    for (auto const & mod: clustColl) {
+        if(mod.empty()) { continue; }
+        DetId detId = mod.id();
+
+        // -- clusters on this det
+        edmNew::DetSet<SiPixelCluster>::const_iterator  di;
+        int nClusterCount=0;
+        for (di = mod.begin(); di != mod.end(); ++di) {
+            nClusterCount++;
+        }
+        int nCluster = mod.size();
+        if(nCluster!=nClusterCount) {
+            std::cout<<"counting yields "<<nClusterCount<<" but the size is "<<nCluster<<"; they should match."<<std::endl;
+        }
+        thePCCob->increment(detId(), bx, nCluster);
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
+    //New PCC object at the beginning of each lumi section
+    thePCCob = std::make_unique<reco::PixelClusterCounts>();
+    
+    if ( countLumi_ == 0 || (resetNLumi_ > 0 && countLumi_%resetNLumi_ == 0) ) {
+        beginLumiOfPCC_ = lumiSeg.luminosityBlock();
+    }
+
+    countLumi_++;
+
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
+    //Saving the PCC object 
+    //FIXME! The line below doesn't work but I need to save here. 
```

Comment:
My personal opinion is you'd probably better be served by writing a summary out each lumi and then have another job which runs over the file to do your final agregation. That would also prepare you for the time when the framework will process multiple luminosity blocks simultaneously (which hopefully will happen by the end of Summer).

As for the beam spot, I checked and all non-test uses of the beam spot producer I could find write each luminosity block. See
https://github.com/cms-sw/cmssw/search?utf8=✓&q=fitEveryNLumi&type=